Sources/Shared/TCPSocket.cpp

```diff
@@ -42,7 +42,9 @@ int TCPSocket::send(const std::string &data) {
   return ::send(this->socketDescriptor, data.c_str(), data.size(), 0);
 }
 
-int TCPSocket::receive(std::string &data) {
+int TCPSocket::receive(std::string &data, int bytes) {
+  char buffer[bytes];
```

Comment:
I don't think it makes a difference at all in this case. 
 It's fully local variable, and I'm aware of the fact that it deprecates to a pointer, but we're not sharing this anywhere. In this case we want block scope and we have it.
Allocation of object on the heap is recommended mainly for long lived objects. Stack allocation is lot faster & we don't have to bother with deallocation.
I think that new and delete here is overkill.